flink-runtime/src/test/java/org/apache/flink/runtime/jobmaster/JobMasterTest.java

```diff
@@ -1384,8 +1374,9 @@ public void testRequestPartitionState() throws Exception {
                 fail("Expected failure.");
             } catch (ExecutionException e) {
                 assertThat(
-                        ExceptionUtils.findThrowable(e, IllegalArgumentException.class).isPresent(),
-                        is(true));
+                                ExceptionUtils.findThrowable(e, IllegalArgumentException.class)
```

Comment:
Would `hasRootCauseInstanceOf(IllegalArgumentException.class)` work here?
If not, we can still simplify it to be `assertThat(ExceptionUtils.findThrowable(e, IllegalArgumentException.class)).isPresent()`.

Reply:
I think this code block can replaced by 
```
Assertions.assertThatThrownBy(() -> unknownPartitionStateFuture.get())
                    .satisfies(FlinkAssertions.anyCauseMatches(IllegalArgumentException.class));
```

Reply:
Another pattern as follow:
```
Assertions.assertThat(partitionStateFuture)
                    .failsWithin(0, TimeUnit.SECONDS)
                    .withThrowableOfType(ExecutionException.class)
                    .satisfies(FlinkAssertions.anyCauseMatches(IllegalArgumentException.class));
```

Reply:
I think `assertThatThrownBy(xxx).hasRootCauseInstanceOf` is enough.